### PR TITLE
Add ignition-transport

### PR DESCRIPTION
**Pull request by joxoby**
Specify library name and version:  **ignition-transport/9.0.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by joxoby** [comment]
Depends on:
[ignition-cmake](https://github.com/conan-io/conan-center-index/pull/3268)
[ignition-msgs](https://github.com/conan-io/conan-center-index/pull/3273)
[protobuf](https://github.com/conan-io/conan-center-index/pull/2037)

File: recipes/ignition-transport/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    @property
+    def _cmake_build_subfolder(self):
+        return "cmake_build_subfolder"
+
+    @property
+    def _cmake_prefix_subfolder(self):
+        return "cmake_prefix_subfolder"
+
+    def source(self):
```
Comment (madebr) [inline recipes/ignition-transport/all/conanfile.py:51]:
```suggestion
        return "source_subfolder"

    def source(self):
```

File: recipes/ignition-transport/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        self.build_requires("pkgconf/1.7.3")
+        self.build_requires("ignition-cmake/2.5.0")
+
+    def _configure_cmake(self):
+        if self._cmake is not None:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.verbose = True
+        self._cmake.definitions["BUILD_TESTING"] = False
+        # self._cmake.definitions["CMAKE_PREFIX_PATH"] = os.path.join(self.build_folder, self._cmake_prefix_subfolder)
```
Comment (madebr) [inline recipes/ignition-transport/all/conanfile.py:102]:
```suggestion
```

File: recipes/ignition-transport/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        # Remove MS runtime files
+        for dll_pattern_to_remove in ["concrt*.dll", "msvcp*.dll", "vcruntime*.dll"]:
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), dll_pattern_to_remove)
+
+    def package_info(self):
+        version_major = tools.Version(self.version).major
+        self.cpp_info.names["cmake_find_package"] = "ignition-transport{}".format(version_major)
+        self.cpp_info.names["cmake_find_package_multi"] = "ignition-transport{}".format(version_major)
+
+        # FIXME: create in file ignition-transport6-all-config.cmake
```
Comment (madebr) [inline recipes/ignition-transport/all/conanfile.py:129]: Have you manually checked the generated `-config.cmake` files?
If there is only one file created, then this FIXME can be removed.

File: recipes/ignition-transport/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        self.cpp_info.components["libignition-transport-all"].libs = []
```
Comment (madebr) [inline recipes/ignition-transport/all/conanfile.py:130]: Is this correct? No libs would mean that this package is header-only.

File: recipes/ignition-transport/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.10.2)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+set(IGN_TRANSPORT_MAJOR_VER "" CACHE STRING "Version of igition-math")
```
Comment (madebr) [inline recipes/ignition-transport/all/test_package/CMakeLists.txt:7]:
```suggestion
set(IGN_TRANSPORT_MAJOR_VER "" CACHE STRING "Version of ignition-transport")
```

File: recipes/ignition-transport/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        self.cpp_info.components["libignition-transport-all"].requires = ["libignition-transport-eigen3"]
```
Comment (madebr) [inline recipes/ignition-transport/all/conanfile.py:131]: These requires do not exist.

**Pull request closed without merging**